.teamcity/builds/Common.kt

```diff
@@ -84,15 +87,20 @@ fun collectArtifacts(buildType: BuildType): BuildType {
   return buildType
 }
 
-fun BuildSteps.commonMaven(init: MavenBuildStep.() -> Unit): MavenBuildStep {
+fun BuildSteps.commonMaven(javaVersion: String, init: MavenBuildStep.() -> Unit): MavenBuildStep {
   val maven =
       this.maven {
         // this is the settings name we uploaded to Connectors project
         userSettingsSelection = "github"
         localRepoScope = MavenBuildStep.RepositoryScope.MAVEN_DEFAULT
 
         dockerImagePlatform = MavenBuildStep.ImagePlatform.Linux
-        dockerImage = "eclipse-temurin:11-jdk"
+        dockerImage =
+            when (javaVersion) {
+              "11" -> "eclipse-temurin:11-jdk"
+              "17" -> "eclipse-temurin:17-jdk"
+              else -> error("Unsupported Java version: $javaVersion")
+            }
```

Comment:
we could define a JavaVersion enum, and derive the image from that and also get rid of this and other similar blocks?

Reply:
yeah, sounds good 👍